main.nf

```diff
@@ -500,26 +483,27 @@ if( params.skip_trimming ){
         file "where_are_my_files.txt"
 
         script:
-        c_r1 = params.clip_r1 > 0 ? "--clip_r1 ${params.clip_r1}" : ''
-        c_r2 = params.clip_r2 > 0 ? "--clip_r2 ${params.clip_r2}" : ''
-        tpc_r1 = params.three_prime_clip_r1 > 0 ? "--three_prime_clip_r1 ${params.three_prime_clip_r1}" : ''
-        tpc_r2 = params.three_prime_clip_r2 > 0 ? "--three_prime_clip_r2 ${params.three_prime_clip_r2}" : ''
+        c_r1 = clip_r1 > 0 ? "--clip_r1 $clip_r1" : ''
+        c_r2 = clip_r2 > 0 ? "--clip_r2 $clip_r2" : ''
+        tpc_r1 = three_prime_clip_r1 > 0 ? "--three_prime_clip_r1 $three_prime_clip_r1" : ''
+        tpc_r2 = three_prime_clip_r2 > 0 ? "--three_prime_clip_r2 $three_prime_clip_r2" : ''
         rrbs = params.rrbs ? "--rrbs" : ''
-        multicore = ''
-        if( task.cpus ){
-            ccore = (((task.cpus as int) - 3) / 3) as int
-            if( ccore > 1 ){
-              multicore = "--cores $ccore"
-            }
+        cores = 1
```

Comment:
```suggestion
        def cores = 1
```

Reply:
I think this is one instance where I know `def` should be used :laughing:

Reply:
Why?

Reply:
Initialising the type of a variable I believe before you give it another value. Probably dont have to but I always do. Let me see if I can dig something up.

Reply:
http://grails.asia/groovy-def-declaration-examples

Reply:
I dont know if its going to have much difference to be honest. Ive just got into the habit of using it and as far as I know its a halfway house to using the correct Groovy implementation in Nextflow.
e.g.
https://github.com/drpatelh/nanoseq/blob/50b2e779d4592739682af658a17423876b2abc07/main.nf#L256-L270

Reply:
There are two other instances of a multicore logic in the pipeline. Maybe they should be adjusted to match this implementation?

https://github.com/nf-core/methylseq/blob/0cae7597d70cdfc41178a3e6428a896be9421c55/main.nf#L571

https://github.com/nf-core/methylseq/blob/0cae7597d70cdfc41178a3e6428a896be9421c55/main.nf#L672

Reply:
Yes, probably - but can we add an issue for this and do it another time? It shouldn't affect the way that the pipeline runs I think and I am keen to get this minor / patch release ASAP (and tomorrow is my last day before christmas holidays 🙄)